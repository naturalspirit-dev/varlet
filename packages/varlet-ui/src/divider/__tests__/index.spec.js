import VarDivider from '../Divider'
import example from '../example'
import { mount } from '@vue/test-utils'

test('test divider example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test divider description prop', () => {
  const text = '文字描述'
  const wrapper = mount(VarDivider, {
    props: {
      inset: 36,
      vertical: true,
      description: text,
      margin: '10',
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})
