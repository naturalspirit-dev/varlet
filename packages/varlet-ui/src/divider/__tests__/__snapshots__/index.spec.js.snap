// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test divider description prop 1`] = `"<div class=\\"var-divider var--box var-divider--vertical var-divider--inset\\" style=\\"height: calc(80% - 36px);\\"><span class=\\"var-divider__text\\">文字描述</span></div>"`;

exports[`test divider examlple 1`] = `
"<div>
  <div class=\\"app-type\\">基本使用</div>
  <p>落霞与孤鹜齐飞，秋水共长天一色。</p>
  <div class=\\"var-divider var--box\\">
    <!--v-if-->
  </div>
  <p>固知一死生为虚诞，齐彭殇为妄作。</p>
</div>
<div>
  <div class=\\"app-type\\">缩进</div>
  <div class=\\"var-divider var--box var-divider--inset\\">
    <!--v-if-->
  </div>
  <div class=\\"var-divider var--box var-divider--inset\\" style=\\"margin: 36px 0px; width: calc(100% - 36px); left: 36px;\\">
    <!--v-if-->
  </div>
  <div class=\\"var-divider var--box var-divider--inset\\" style=\\"width: calc(100% - 36px); left: 0px;\\">
    <!--v-if-->
  </div>
</div>
<div>
  <div class=\\"app-type\\">垂直分割线</div>
  <div class=\\"vertical-divider-wrapper\\" style=\\"height: 32px;\\"><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--disabled var-button--text-default var-button--text var-button--text-disabled\\" disabled=\\"\\">
      <!--v-if-->
      <div class=\\"var-button__content\\">文字</div>
    </button>
    <div class=\\"var-divider var--box var-divider--vertical\\">
      <!--v-if-->
    </div><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-primary var-button--text\\">
      <!--v-if-->
      <div class=\\"var-button__content\\">链接</div>
    </button>
    <div class=\\"var-divider var--box var-divider--vertical\\">
      <!--v-if-->
    </div><button class=\\"var-button var--box var-button--normal var--inline-flex var-button--text-default var-button--text\\">
      <!--v-if-->
      <div class=\\"var-button__content\\"><i class=\\"var-icon var-icon--set var-icon-github\\" style=\\"transition: all 0ms;\\"></i></div>
    </button>
  </div>
</div>
<div>
  <div class=\\"app-type\\">带有文字描述的分割线</div>
  <div class=\\"var-divider var--box\\"><span class=\\"var-divider__text\\">文字描述</span></div>
</div>
<div>
  <div class=\\"app-type\\">自定义</div>
  <div class=\\"var-divider var--box\\"><i class=\\"var-icon var-icon--set var-icon-heart-outline\\" style=\\"color: rgb(41, 121, 255); transition: all 0ms; margin: 0px 16px;\\"></i></div>
</div>"
`;
